tests/commands/test_changelog_command.py

```diff
@@ -20,10 +21,12 @@ def test_changelog_on_empty_project(mocker):
     testargs = ["cz", "changelog", "--dry-run"]
     mocker.patch.object(sys, "argv", testargs)
 
-    with pytest.raises(NoCommitsFoundError) as excinfo:
+    with pytest.raises(GitCommandError):
         cli.main()
 
-    assert "No commits found" in str(excinfo)
+    #  git will error out with something like:
+    #  "your current branch 'XYZ' does not have any commits yet"
+    #  is it wise to assert the exception contains a message like this?
```

Comment:
If we need to compare the error message of external tool (e.g., `git`), we'll need to take `LC_ALL` into consideration. Otherwise, people with difference `LC_ALL` will not pass the test

Reply:
We could try something like changing the git commit command to `LC_ALL=en_US.UTF-8 git commit`

Reply:
you bring up a really good point. Are you happy as-is? How would you like to proceed with this PR?

Reply:
I notice you join the discussion [here](https://github.com/commitizen-tools/commitizen/pull/543#issuecomment-1203038161). Maybe we could discuss this kind of issue there 🤔 

> How would you like to proceed with this PR?

Yep, I'm ok with asserting the message. 

> How would you like to proceed with this PR?

We'll need a way to resolve the language issue. After that, I think we're almost ready to merge

Reply:
What about this approach of retrying the command?

1. `execinfo` captures the command that failed.
2. test retries executing the command with `LC_ALL`.
3. assert the retry matches the expected `en_US` output of the command.

``` python
    with pytest.raises(GitCommandError) as execinfo:
         cli.main()

     c = cmd.run("LC_ALL=en_US.UTF-8 " + execinfo.value.cmd)
     assert "does not have any commits yet" in c.err
```

Reply:
I'm good with this approach :) 

@Woile what do you think. Different from what's discussed on #543  `LC_ALL` part only happens in test cases.

Reply:
I wonder if I should have just added `LC_ALL=en_US.UTF-8` to the entire testing environment.

Reply:
that would be quite a bit more straightforward than this retry strategy.

Reply:
assuming we don't mock out that specific git call entirely.

Reply:
If it's just for testing, we could use [pytest-env](https://pypi.org/project/pytest-env/) to achieve it. If it's for the functional code, maybe we'll need to think of a better solution like we did on #543

Reply:
@Lee-W After much thought, this test probably isn't providing much on the `command` level.
I think all we want to be concerned about is that we handle an empty git log properly. 

I have refactored `get_commits` so that it calls a `get_log` function that we can test as it's own unit. https://github.com/commitizen-tools/commitizen/pull/540/commits/4317d9f3aba9cb5edba702f0076155118578fad1

Reply:
As in:

**tests/test_git.py**
``` python
@pytest.mark.usefixtures("tmp_commitizen_project")
 def test_get_log_as_str_list_empty():
     """ensure an exception or empty list in an empty project"""
     try:
         gitlog = git._get_log_as_str_list(start=None, end="HEAD", args="")
     except exceptions.GitCommandError:
         return
     assert len(gitlog) == 0, "list should be empty if no assert"
```

replaces: <br>

**tests/commands/test_changelog_command.py**
``` python
@pytest.mark.usefixtures("tmp_commitizen_project")
 def test_changelog_on_empty_project(mocker):
     testargs = ["cz", "changelog", "--dry-run"]
     mocker.patch.object(sys, "argv", testargs)

     with pytest.raises(GitCommandError):
         cli.main()

     #  git will error out with something like:
     #  "your current branch 'XYZ' does not have any commits yet"
     #  is it wise to assert the exception contains a message like this?
```

Reply:
And then no need for the `LC_ALL` at all since the function being tested is so small in comparison to before when we tested the whole command